[CI/CD] Actuated arm builds intermittently fail to install dependencies

### Component(s)

_No response_

### Describe the issue you're reporting

[Example 1](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/8648630092/job/23712986771)
[Example 2](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/8649711039/job/23716887288)

Failure output:
```
Running target 'moddownload' in module 'receiver/apachereceiver' as part of group 'all'
make --no-print-directory -C receiver/apachereceiver moddownload
go mod download
go: github.com/lufia/plan9stats@v0.0.0-20211012122336-39d0f177ccd0: Get "https://proxy.golang.org/github.com/lufia/plan9stats/@v/v0.0.0-20211012122336-39d0f177ccd0.zip": unexpected EOF
make[2]: *** [../../Makefile.Common:221: moddownload] Error 1
make[1]: *** [Makefile:165: receiver/apachereceiver] Error 2
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory '/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib'
make: *** [Makefile:109: gomoddownload] Error 2
Error: Process completed with exit code 2.
```

```
Running target 'moddownload' in module 'receiver/datadogreceiver' as part of group 'all'
make --no-print-directory -C receiver/datadogreceiver moddownload
go: downloading github.com/DataDog/datadog-agent/pkg/proto v0.51.1-0.20240301173728-334e775e420a
go mod download
go: github.com/vmihailenco/msgpack/v4@v4.3.13: Get "https://proxy.golang.org/github.com/vmihailenco/msgpack/v4/@v/v4.3.13.mod": read tcp 192.168.128.33:[592](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/8648630092/job/23712986771#step:8:593)40->142.250.113.141:443: read: connection reset by peer
make[2]: *** [../../Makefile.Common:221: moddownload] Error 1
make[1]: *** [Makefile:165: receiver/datadogreceiver] Error 2
make[1]: *** Waiting for unfinished jobs....
go mod download
make[1]: Leaving directory '/home/runner/actions-runner/_work/opentelemetry-collector-contrib/opentelemetry-collector-contrib'
make: *** [Makefile:109: gomoddownload] Error 2
Error: Process completed with exit code 2.
```

--- a/.github/workflows/build-and-test.yml
+++ b/.github/workflows/build-and-test.yml
@@ -8,6 +8,7 @@ on:
   pull_request:
 env:
   TEST_RESULTS: testbed/tests/results/junit/results.xml
+  GOPROXY: https://goproxy.cncf.actuated.dev,direct
   # Make sure to exit early if cache segment download times out after 2 minutes.
   # We limit cache download as a whole to 5 minutes.
   SEGMENT_DOWNLOAD_TIMEOUT_MINS: 2

